src/rules/group-exports.js

```diff
@@ -0,0 +1,133 @@
+const meta = {}
+/* eslint-disable max-len */
+const errors = {
+  ExportNamedDeclaration: 'Multiple named export declarations; consolidate all named exports into a single statement',
+  ExportDefaultDeclaration: 'Default export declaration should be adjacent to named export',
+  MemberExpression: 'Multiple CommonJS exports; consolidate all exports into a single assignment to `module.exports`',
+}
+/* eslint-enable max-len */
+
+/**
+ * Check if two nodes are adjacent (only whitespace between them)
+ *
+ * The two nodes do not have to be sorted in the order they appear in the code.
+ *
+ * @param     {Object}    opts            Options for the check
+ * @param     {Object}    opts.context    The context of the nodes
+ * @param     {Object}    opts.first      The first node
+ * @param     {Object}    opts.second     The second node
+ * @return    {Boolean}
+ * @private
+ */
+function isAdjacent(opts = {}) {
+  const sourceCode = opts.context.getSourceCode()
+
+  if (sourceCode.getTokensBetween(opts.first, opts.second).length ||
+    sourceCode.getTokensBetween(opts.second, opts.first).length) {
+    return false
+  }
+
+  return true
+}
+
+/**
+ * Determine how many property accesses precede this node
+ *
+ * For example, `module.exports` = 1, `module.exports.something` = 2 and so on.
+ *
+ * @param     {Object}    node    The node being visited
+ * @return    {Number}
+ * @private
+ */
+function accessorDepth(node) {
+  let depth = 0
+
+  while (node.type === 'MemberExpression') {
+    depth++
+    node = node.parent
+  }
+
+  return depth
+}
+
+function create(context) {
+  const exports = {
```

Comment:
Can you rename this to something else? `exports` is what I consider a "semi-reserved" keyword in Node, better to name it something else to avoid confusion.